src/clj/fluree/db/query/exec/select.cljc

```diff
@@ -52,16 +53,19 @@
 
 (defprotocol ValueSelector
   (implicit-grouping? [this]
-   "Returns true if this selector should have its values grouped together.")
+    "Returns true if this selector should have its values grouped together.")
   (format-value [fmt db iri-cache context compact fuel-tracker error-ch solution]
-   "Formats a where search solution (map of pattern matches) by extracting and displaying relevant pattern matches."))
+    "Async format a search solution (map of pattern matches) by extracting relevant match.")
+  (solution-value [fmt error-ch solution]
```

Comment:
I feel like this should be in a new subquery specific protocol because the solution-value method doesn't apply in all the cases where ValueSelector should be defined.

Reply:
I started this change, but both `format-value` and `solution-value` are tied to `implicit-grouping?`. So either `solution-value` needs to rely on the other protocol. At that point it seems they should be part of the same protocol.

Do you still want them separated out?

Reply:
Yeah, I have had reservations about the addition of `implicit-grouping?` as a protocol method, and especially as a method on _this_ protocol since it was introduced because whether or not something is grouped implicitly is outside the scope of selecting values. 

I think the best thing would be to [go back to the way it was before](https://github.com/fluree/db/blob/e21b2039bb3a086510c378634115b1f0dac43ebd/src/fluree/db/query/exec/select.cljc#L137-L140) and just add the other aggregate selector types to that function as necessary. That would eliminate problems like this and result in more flexibility. 

I think that's probably beyond the scope of this particular pr so I might just make that change along with the changes I suggested here in a separate cleanup branch at some point.

Reply:
I don't plan to resolve this here - I don't think it is specific to the subquery functionality and can sit as a separate change. We are also still doing some query work, so opinions on ideal solution might change.